runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -1101,6 +1101,33 @@ inlineDoubleMin(TR::Node *node, TR::CodeGenerator *cg)
    return doubleMaxMinHelper(node, cg, false);
    }
 
+extern TR::Register * 
+inlineMathFma(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT_FATAL(node->getNumChildren() == 3, 
+   "In function inlineMathFma, the node at address %p should have exactly 3 children, but got %u instead", node, node->getNumChildren());
+
+   TR::Register      * targetRegister      = cg->allocateRegister(TR_VRF);  
+
+   TR::Register      * v1      = cg->evaluate(node->getFirstChild());
+   TR::Register      * v2      = cg->evaluate(node->getSecondChild());
+   TR::Register      * v3      = cg->evaluate(node->getThirdChild());
+
+   uint8_t mask6 = getVectorElementSizeMask(TR::DataType::getSize(node->getDataType()));
+   generateVRReInstruction(cg, TR::InstOpCode::VFMA, node, targetRegister, v1, v2, v3, mask6, 0);
+
+   cg->stopUsingRegister(v1);
+   cg->stopUsingRegister(v2);
+   cg->stopUsingRegister(v3);
```

Comment:
This is a bug. It will prematurely kill the registers associated with the node. If the children of the call are commoned somewhere then this could result in a garbage value being held in the corresponding register at runtime. We need to remove these three lines. In general we only call `stopUsingRegister` on registers which were allocated within the evaluator or helper (i.e. registers created via `allocateRegister` API).

Can we add some unit tests for this API? I'd like to see a commoning test in there which would have caught this bug.

Reply:
I will remove these lines.
I will look at how we can add unit tests to prevent this kind of bug.

Reply:
Here is an example for some math methods that the JIT recognizes:
https://github.com/eclipse/openj9/blob/master/test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java

You can likely create a similar test class and write some unit tests there.